packages/ketchup/src/components/kup-activity-timeline/kup-activity-timeline.tsx

```diff
@@ -0,0 +1,433 @@
+                ...this.getActivity(activityData),
+                originalEvent: event,
+                type: KupActivityTimelineAction.onRightClick,
+            },
+        });
+    }
+
+    activityItem(activity: KupActivity, activityIndex: number) {
+        return (
+            <div key={activityIndex}>
+                <table class="atm-detail">
+                    <tbody>
+                        {activity.columns.map(
+                            (
+                                columnChunk: KupActivityTimelineData[],
+                                chunkIndex: number
+                            ) => (
+                                <tr key={chunkIndex}>
+                                    {columnChunk.map(
+                                        (column: KupActivityTimelineData) => (
+                                            <Fragment>
+                                                <td class="detail-label">
+                                                    {column.title} :
+                                                </td>
+                                                <td class="detail-value">
+                                                    <a
+                                                        href="javascript:void(0)"
+                                                        class="ui-commandlink ui-widget"
+                                                        onClick={(e) => {
+                                                            e.stopPropagation();
+                                                            this.onActivityClick(
+                                                                e,
+                                                                column
+                                                            );
+                                                        }}
+                                                        onContextMenu={(e) => {
+                                                            e.stopPropagation();
+                                                            this.onActivityContextMenu(
+                                                                e,
+                                                                column
+                                                            );
+                                                        }}
+                                                    >
+                                                        {column.value}
+                                                    </a>
+                                                </td>
+                                            </Fragment>
+                                        )
+                                    )}
+                                </tr>
+                            )
+                        )}
+                    </tbody>
+                </table>
+                <br />
+            </div>
+        );
+    }
```

Comment:
The piece of code that create the table is not bad but it's not responsive, if you shrink the ATM you'll notice that it doesn't display correctly. What i suggest you is to abort the table and use the CSS Grid to display the table element and the CSS Flex for the Cells. In that way, with simple css, you can display elements on 1, 2 or 4 cols based on the screen width